### PR TITLE
OME Tiff - Deprecating getMetadataStoreForConversion and Display

### PR DESCRIPTION
**Pull request by dgault**
This is a follow up PR to https://github.com/openmicroscopy/bioformats/pull/2502

In this PR I have made the following changes based on discussions and feedback:
 - Deprecated the getMetadataStoreForConversion and getMetadataStoreForDisplay  methods in the OMETiffReader as done in ome/ome-files-cpp@5bc636b
- Dropped the call to getMetadataForConversion in OMETiffReader
- Dropped the call oto getMetadataForDisplay in ImageInfo 
- When using using showinf with the --omexml flag is set then the validation flag will also be set to true (unless the novalid flag is also set in which case it will override)

### CODE REVIEW

File: components/formats-bsd/src/loci/formats/in/OMETiffReader.java [hunk 1]
```diff
@@ -1026,49 +1026,45 @@ else if (m.sizeC == m.imageCount) {
...
-   * Returns a MetadataStore that is populated in such a way as to
-   * produce valid OME-XML.  The returned MetadataStore cannot be used
-   * by an IFormatWriter, as it will not contain the required
-   * BinData.BigEndian attributes.
+   * Get a MetadataStore suitable for display.
+   *
+   * @note Historically, this method removed certain elements
+   * for display purposes and was not be suitable for use with
+   * FormatWriter due to not containing required BinData
+   * BigEndian attributes pthis requirement has long been
```
Comment (mtbc) [inline components/formats-bsd/src/loci/formats/in/OMETiffReader.java]: "pthis"

File: components/formats-bsd/src/loci/formats/in/OMETiffReader.java [hunk 2]
```diff
@@ -1026,49 +1026,45 @@ else if (m.sizeC == m.imageCount) {
...
+   * unnecessary].  This is no longer the case; the general
+   * FormatReader::getMetadataStore() method will always create
+   * valid metadata which is suitable for both display and use
+   * with FormatWriter, and so should be used instead.
+   *
+   * @returns the metadata store.
+   * @deprecated Use the general FormatReader::getMetadataStore() method.
```
Comment (mtbc) [inline components/formats-bsd/src/loci/formats/in/OMETiffReader.java]: `@link`?

File: components/formats-bsd/src/loci/formats/in/OMETiffReader.java [hunk 3]
```diff
@@ -1026,49 +1026,45 @@ else if (m.sizeC == m.imageCount) {
...
-   * Returns a MetadataStore that is populated in such a way as to
-   * produce valid OME-XML.  The returned MetadataStore cannot be used
-   * by an IFormatWriter, as it will not contain the required
-   * BinData.BigEndian attributes.
+   * Get a MetadataStore suitable for display.
+   *
+   * Note: Historically, this method removed certain elements
+   * for display purposes and was not be suitable for use with
+   * FormatWriter due to not containing required BinData
+   * BigEndian attributes pthis requirement has long been
```
Comment (mtbc) [inline components/formats-bsd/src/loci/formats/in/OMETiffReader.java]: "pthis"

File: components/formats-bsd/src/loci/formats/in/OMETiffReader.java [hunk 4]
```diff
@@ -1026,49 +1026,45 @@ else if (m.sizeC == m.imageCount) {
...
+   * unnecessary].  This is no longer the case; the general
+   * FormatReader::getMetadataStore() method will always create
+   * valid metadata which is suitable for both display and use
+   * with FormatWriter, and so should be used instead.
+   *
+   * @return the metadata store.
+   * @deprecated Use the general FormatReader::getMetadataStore() method.
```
Comment (mtbc) [inline components/formats-bsd/src/loci/formats/in/OMETiffReader.java]: `{@link ... }` and elsewhere

**Comment by sbesson** [comment]
From the tests point of view, this PR seems to work as expected as shown by https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-repository-subset/269.

File: components/bio-formats-tools/src/loci/formats/tools/ImageInfo.java [hunk 5]
```diff
@@ -204,7 +204,10 @@ public boolean parseArgs(String[] args) {
         else if (args[i].equals("-separate")) separate = true;
         else if (args[i].equals("-expand")) expand = true;
         else if (args[i].equals("-cache")) cache = true;
-        else if (args[i].equals("-omexml")) omexml = true;
+        else if (args[i].equals("-omexml")) {
+          omexml = true; 
+          validate = true;
```
Comment (sbesson) [inline components/bio-formats-tools/src/loci/formats/tools/ImageInfo.java]: The main issue with setting both parameters in this block is that it makes the order of the arguments matter, i.e. the two following commands do not have the same functionality:

```
./showinf  -nopix -novalid -omexml file.ome.tiff
./showinf  -nopix -omexml -novalid file.ome.tiff 
```

As `validate` is conditional to `omexml`, thoughts about setting the default value to `true` and leaving `-validate/-novalid` to override it as in `ome-files info`?

**Comment by sbesson** [comment]
The changes in this PR make sense, the deprecation/Javadoc explanations read well to me and the OME-TIFF samples pass the [automated tests](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-repository-subset/271).

Also tested the OME-TIFF to OME-TIFF conversion to check the impact of dropping `getMetadataStoreForConversion` in OMETiffReader using 2013-06 samples files (single file, multi-file, binary only, modulo) and a few OME-TIFF from our curated data repository. Everything worked as expected with no error produced during the conversion and the generated files re-open with `showinf` as expected.

Once https://github.com/openmicroscopy/bioformats/pull/2507#discussion_r73336709 is cleared, this looks good to merge.

**Comment by dgault** [comment]
Having it on by default makes sense to me. The omexml has been updated so it no longer sets validate. The validation is defaulting to true unless overridden by  -validate/-novalid

**Comment by sbesson** [comment]
With the last commit, everything works as expected. Merging.

**Pull request merged**